src/ripple/rpc/impl/LookupLedger.cpp

```diff
@@ -83,46 +64,66 @@ Status ledgerFromRequest (
         uint256 ledgerHash;
         if (! ledgerHash.SetHex (hashValue.asString ()))
             return {rpcINVALID_PARAMS, "ledgerHashMalformed"};
-           
+
         ledger = netOps.getLedgerByHash (ledgerHash);
+        if (ledger == nullptr)
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
     }
     else if (indexValue.isNumeric())
     {
         ledger = netOps.getLedgerBySeq (indexValue.asInt ());
-    }
-    else
-    {
-        auto const index = indexValue.asString ();
-        auto const isCurrent = index.empty() || index == "current";
-        if (isCurrent)
-            ledger = netOps.getCurrentLedger ();
-        else if (index == "closed")
-            ledger = netOps.getClosedLedger ();
-        else if (index == "validated")
-            ledger = netOps.getValidatedLedger ();
-        else
-            return {rpcINVALID_PARAMS, "ledgerIndexMalformed"};
-
         if (ledger == nullptr)
-            return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+            return {rpcLGR_NOT_FOUND, "ledgerNotFound"};
 
-        assert (ledger->isImmutable());
-        assert (ledger->isClosed() == !isCurrent);
+        if (ledger->getLedgerSeq () > netOps.getValidatedSeq () &&
+            isValidatedOld ())
+        {
+            return {rpcNO_NETWORK, "InsufficientNetworkMode"};
+        }
```

Comment:
The user asked for a ledger by number from the future, and rippled managed to find one. If we are out of sync (`isValidatedOld()`), the ledger is meaningless, so don't give it to them.